src/auto-evo/RunResults.cs

```diff
@@ -368,6 +328,155 @@ public long GetPopulationInPatch(Species species, Patch patch)
             }
         }
 
+        /// <summary>
+        ///   Computes the final population of all species, by patch.
+        /// </summary>
+        public Dictionary<Patch, Dictionary<Species, long>> GetPopulationsByPatch(bool resolveMigrations = false,
+            bool resolveSplits = false, bool includeNewSpecies = true)
+        {
+            var speciesInPatches = new Dictionary<Patch, Dictionary<Species, long>>();
+            foreach (var species in results.Keys)
+            {
+                var populations = GetPopulationsByPatch(species, resolveMigrations, resolveSplits, includeNewSpecies);
+                foreach (var patchEntry in populations)
+                {
+                    if (!speciesInPatches.ContainsKey(patchEntry.Key))
+                        speciesInPatches[patchEntry.Key] = new Dictionary<Species, long>();
+
+                    patchEntry.Value.ToList().ForEach(s =>
+                        speciesInPatches[patchEntry.Key].Add(s.Key, s.Value));
+                }
+            }
+
+            return speciesInPatches;
+        }
+
+        /// <summary>
+        ///   Computes the final population of a species and its splitted forms, by patch.
+        /// </summary>
+        /// <remarks>
+        ///   <para>
+        ///     Species are only returned if their population is above 0.
+        ///   </para>
+        /// </remarks>
+        public Dictionary<Patch, Dictionary<Species, long>> GetPopulationsByPatch(Species species,
+            bool resolveMigrations = false, bool resolveSplits = false, bool includeNewSpecies = true)
+        {
+            var speciesInPatches = new Dictionary<Patch, Dictionary<Species, long>>();
+
+            if (!results.ContainsKey(species))
+            {
+                GD.Print("Species ", species.FormattedName, "not found in results, returning empty dictionary.");
```

Comment:
```suggestion
                GD.PrintErr("Species ", species.FormattedName, "not found in results, returning empty dictionary.");
```
I'd go even as far as to say that I think this should throw an exception. There's a reason I think in the original code I had exception throwing, it's a bad logic error to try to read the details of a species that wasn't included in the simulation.